index.js

```diff
@@ -1,44 +1,149 @@
     })
+    .catch(err => {
+        // res.status(500).send(`<h1>Bad Request<h1>`);
+        res.status(500).json({message: `failed to get users`});
+        //res.(status(500).json(err);
+    });
 });
     
 
-server.get('/api/posts/:id', (req,res) => {
+server.get('/api/posts/:id', (req, res) => {
     // console.log(req);
     console.log(req.params);
-    const thisId = req.params.id;
-    db.findById(thisId)
-        .then( thisPost => {
-            console.log(`thisPost = ${thisPost}`);
-            res.status(200).send(thisPost);
+    const id = req.params.id;
+
+    db.findById(id)
+        .then( posts => {
+            console.log(`posts = ${posts.id}`);
+            res.status(200).json(posts);
         })
-        .catch(
-            res.status(404).json({ message: `User does not exist.`})
-        )
-})
-//     .then()
-//     .catch();
+        .catch(err => {
+            res.status(500).json({ message: `User does not exist.`});
+        });
+    
+});
+
+//++++++++++++++++++++++++++++++++++++++++
+// Day 2 - put post delete stuff here
+//++++++++++++++++++++++++++++++++++++++++
+
+
+server.post('/api/posts', (req, res) => {
+    const post = req.body;
+    console.log(req.body);
+    if (post.title && post.contents) {
+        db.insert(post)
+            .then( postData => {
+                console.log('post from body', postData);
+                db.findById(postData.id).then( post => {
+                    res.status(201).json(post);
+                });
+            })
+            .catch( (err) => {
+                res.status(500)
+                    .json({
+                    message: `Failed to insert user into the db`
+                })
+            });
+
+    } else {
+        console.log("++++ERROR missing title or contents!! +++");
+        res.status(400).json({ message: "missing manditory data!!"});
+    }
+});
+
+
 
 
 
-server.listen(PORT, () => {
-    console.log(`The server is runnning on port ${PORT}`);
-});
\ No newline at end of file
+//++++++++++++++++++++++++++++++++++++
+//+++++++  Put  ++++++++++++ regular Put
+//++++++++++++++++++++++++++++++++++++++
+
+server.put("/api/posts/:id", (req, res) => {
+    const { id } = req.params;
+    const changes = req.body;
+  
+    db.findById(id)
+      .then(post => {
+        if (!post) {
+          res
+            .status(404)
+            .json({ message: "The post with the specified ID does not exist." });
+        }
+        if (!changes.title || !changes.contents) {
+          res.status(400).json({
+            errorMessage: "Please provide title and contents for the post."
+          });
+        }
+        db.update(id, changes).then(newPost => {
+          res.status(200).json(newPost);
+        });
+      })
+      .catch(err =>
+        res
+          .status(500)
+          .json({ error: "The post information could not be modified." })
+      );
+  });
+
+
+server.delete("/api/posts/:id", async (req, res) => {
+    const id = req.params.id;
```

Comment:
You _could_ destructure the id here; this is a stylistic suggestion over anything else.